### PR TITLE
Improve Redis connection hygiene

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR aims to improve our Redis connection hygiene. Currently, we create a new Redis connection for every command. Since Redis is naturally single-threaded, this approach does not offer any significant performance benefits and instead accumulates connections on our Redis node.

The proposed changes in this PR establish a single connection for all Redis operations, with the exception of blocking streaming commands (like `xRead`). These commands do not block Redis itself but do block the client, so they will run on an isolated pool. We have set up a pool of 200 connections per pod to handle our streaming needs, which should be sufficient and can be monitored through our Datadog (DD) dashboard. Presently, one conversation consumes up to 2 Redis connections when streaming, meaning that with the current setup, we can support up to 100 conversations per minute per pod. This capacity is sufficient for our current usage (see [here](https://dust.tt/w/0ec9852c2f/assistant/kjc7Wa14a4)).

These isolated connections will be automatically closed once they complete their tasks ([documentation](https://github.com/redis/node-redis/blob/HEAD/docs/isolated-execution.md)).

We no longer call `.quit()` on the connections. I will follow up with a PR to handle signals to ensure proper cleanup of Redis connections. For now, if a Kubernetes pod dies, all connections managed by the pod will be terminated, which should be safe enough.

## Risk

In the worst-case scenario, it could disrupt our streaming logic, but our monitoring system should detect this promptly. If necessary, we can safely roll back the changes. 

## Deploy Plan

- [x] Test on front-edge

### CODE REVIEW

File: front/lib/api/assistant/agent.ts [hunk 1]
```diff
@@ -43,9 +43,9 @@ import {
   renderConversationForModelMultiActions,
```
Comment (flvndvd) [inline front/lib/api/assistant/agent.ts:43]: Better reviewed with hidden whitespace ([link](https://github.com/dust-tt/dust/pull/6187/files?diff=split&w=1)).

File: front/lib/api/redis.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import type { RedisClientType } from "redis";
```
Comment (flvndvd) [inline front/lib/api/redis.ts:1]: Moved from the `front/redis` since it does belong to back end helpers.

File: front/lib/api/redis.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    client = createClient({
+      url: REDIS_URI,
+      isolationPoolOptions: {
+        acquireTimeoutMillis: 10000, // 10 seconds.
+        // We support up to 200 concurrent connections for streaming.
+        max: 200,
+      },
+    });
+    client.on("error", (err) => logger.info("Redis Client Error", err));
```
Comment (lasryaric) [inline front/lib/api/redis.ts:23]: We need to enforce the logging format here:
```
logger.info({err}, "Redis Client Error");
```

**Review by lasryaric — APPROVED** [review]
LGTM minus one little comment.
This definitely looks like the right way to do it 👍

**Review by spolu — APPROVED** [review]
Looks reasonable assuming this is well understood and tested including concurrent streaming of conversations :+1:

File: front/lib/api/redis.ts (hunk 3 above)
Comment (flvndvd) [reply front/lib/api/redis.ts]: Done.

**Pull request merged**